sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/JsonMessageCodec.scala

```diff
@@ -82,18 +81,43 @@ private[kalix] class JsonMessageCodec extends MessageCodec {
   }
 
   private def computeTypeHint(clz: Class[_]): TypeHint = {
-    val typeName = Option(clz.getAnnotation(classOf[TypeName]))
-      .collect { case ann if ann.value().trim.nonEmpty => ann.value() }
-      .getOrElse(clz.getName)
+    if (clz.getName.contains("java.lang")) {
+      val typeHint = if (clz.isAssignableFrom(classOf[String])) {
+        TypeHint("string", List("string", "java.lang.String"))
+      } else if (clz.isAssignableFrom(classOf[Integer])) {
+        TypeHint("int", List("int", "java.lang.Integer"))
+      } else if (clz.isAssignableFrom(classOf[Long])) {
+        TypeHint("long", List("long", "java.lang.Long"))
+      } else if (clz.isAssignableFrom(classOf[Boolean])) {
+        TypeHint("boolean", List("boolean", "java.lang.Boolean"))
+      } else if (clz.isAssignableFrom(classOf[Double])) {
+        TypeHint("double", List("double", "java.lang.Double"))
+      } else if (clz.isAssignableFrom(classOf[Float])) {
+        TypeHint("float", List("float", "java.lang.Float"))
+      } else if (clz.isAssignableFrom(classOf[Character])) {
+        TypeHint("char", List("char", "java.lang.Character"))
+      } else if (clz.isAssignableFrom(classOf[lang.Byte])) {
+        TypeHint("byte", List("byte", "java.lang.Byte"))
+      } else if (clz.isAssignableFrom(classOf[Short])) {
+        TypeHint("short", List("short", "java.lang.Short"))
+      } else {
+        TypeHint(clz.getName, List(clz.getName))
+      }
+      typeHint.allTypeHints.foreach(className => addToReversedCache(clz, className))
```

Comment:
Whoever designed this codec must be a genius. `allTypeHints` was already to cover this kind of situation! 😄